SwiftCompilerSources/Sources/SIL/KeyPath.swift

```diff
@@ -0,0 +1,285 @@
+//===--- KeyPath.swift ----------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2025 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import AST
+import SILBridging
+
+//============================================================================//
+// KeyPathPattern
+//============================================================================//
+
+public struct KeyPathPattern {
+  public let bridged: BridgedKeyPathPattern
+
+  public init(bridged: BridgedKeyPathPattern) {
+    self.bridged = bridged
+  }
+
+  public init(
+    _ function: Function,
+    _ rootTy: CanonicalType,
```

Comment:
naming conventions: When you make `rootTy` an argument label, it should be spelled `rootType`. Abbreviations should be avoided